src/actions/peaks-instance.js

```diff
@@ -7,3 +7,40 @@ export function initPeaks(smData, options) {
     options
   };
 }
+
+export function insertNewSegment(smData) {
+  return {
+    type: types.INSERT_SEGMENT,
+    payload: smData
```

Comment:
Is this the entire `smData`, or just the new segment array or object data?

Reply:
This is the entire thing. First I passed only the new span, but it not possible to rearrange the colors with only new timespan. Because the segments array in the peaks instance are not ordered. On second thought, I can sort it by the start times and get the order. What do you think? Remove passing the entire thing and use the peaks instance?

Reply:
Yea, I think we make the action payload data as small as possible, where it's possible.  It might not matter for this test data, but if there's some huge 100 list data structure, it could get a little sluggish.

Reply:
Yeah, I agree. It might slow things. I will look into this and see what can be changed.